### ISSUE TITLE
Everware hangs when trying to clone from "git@"

### ISSUE THREAD

**Issue by kazeevn** [everware/everware#77]
Steps to reproduce:
1. Paste  git@github.com:yandexdataschool/reproducible_analysis_course.git into repository URL
2. It shows "Cloning repository git@github.com:yandexdataschool/reproducible_analysis_course.git" with no signs of progress.

**Comment by anaderi** [comment]
can you check if https link works?

**Comment by betatim** [comment]
This is a known problem. It gets stuck because `ssh` is asking for a human to confirm the identity of github.com's fingerprint. We need to find a way to disable this check.

**Comment by betatim** [comment]
This is also a regression as we had some JS to convert `git@github.com:...` URLs pasted into the repository URL form into their `https://` equivalent and informing the user of it.

**Comment by ibab** [comment]
I have the same problem with the public instance at the moment.
Stuck at

```
Cloning repository https://github.com/betatim/everware-demo
```

**Comment by betatim** [comment]
This must be a different problem though no? You put in a https:// URL. I can confirm that it seems to hang right now.

**Comment by betatim** [comment]
@kazeevn would you be interested in adding some JS to our HTML form that informs the user that `git@github.com..` isn't supported right now and then transforms it into the `https://...` equivalent?

We could find the JS that we once had in the history of the repository and use that as a starting point.

**Comment by kazeevn** [comment]
@betatim, I think it's better to fix the problem than to code workarounds. It looks like we need only a small modification of .ssh/config
https://serverfault.com/questions/132970/can-i-automatically-add-a-new-host-to-known-hosts

If you want to disable git@ repositories - I suggest doing server-side, as users can pass parameters directly in the url without the JS.

**Comment by betatim** [comment]
You are right we should also raise an exception when the user explicitly passes a ssh URL instead of hanging.

Unfortunately we don't control the ssh config of a container. Can we pass command line options to `git` that influence the behaviour of `ssh`?

**Comment by betatim** [comment]
The JS is mainly to help users who accidentally/unknowingly enter a `git@...` URL and expect it to work.

**Comment by betatim** [comment]
This is the JS used previously to inform the user that only `https://` style URLs are valid. In addition we want to raise an exception in the [form handling part of the spawner](https://github.com/everware/everware/blob/master/everware/spawner.py#L136) if someone submits a `git@` URL anyway.

``` js
{% block script %}
<script type="text/javascript">
require(["home"]);

function replaceRepoURL(e) {
    var target = e.target || e.srcElement
    var repo_url = target.repo_input.value;

    if (repo_url.startsWith('git@')) {
      e.preventDefault();

      repo_url = repo_url.replace(':', '/').replace('git@', 'https://');
      target.repo_input.value = repo_url;

      var msg = document.createElement("div");
      msg.innerText = "We changed the repository's URL for you.";
      var form_div = document.getElementById('form_div');
      console.log(form_div);
      form_div.insertBefore(msg, document.getElementById("repourl_div"));
    }
}

var form = document.getElementById('repo_url_entry');
if (form.attachEvent) {
  form.attachEvent("submit", replaceRepoURL);
} else {
  form.addEventListener("submit", replaceRepoURL);
}

</script>
{% endblock %}
```

**Comment by anaderi** [comment]
why don't you PR it? )

**Comment by betatim** [comment]
Wasn't sure if @kazeevn wanted to pick this up? It would make a good first contribution.

**Comment by betatim** [comment]
This is being partly addressed in #105
